irods/client_init.py

```diff
@@ -53,55 +50,63 @@ def write_native_credentials_to_secrets_file(password, overwrite=True, **kw):
     _write_encoded_auth_value(auth_file, password, overwrite)
 
 
-def write_pam_credentials_to_secrets_file(password, overwrite=True, **kw):
+# Reverse compatibility.
+write_native_credentials_to_secrets_file = write_native_irodsA_file
+
+
+# Facility for writing authentication secrets file.  Designed to be useable for iRODS 4.3(+) and 4.2(-).
+def write_pam_irodsA_file(password, overwrite=True, ttl="", **kw):
+    import irods.auth.pam_password
+    from irods.session import iRODSSession
+    import io
+
+    ses = kw.pop("_session", None) or h.make_session(**kw)
+    if ses._server_version(iRODSSession.GET_SERVER_VERSION_WITHOUT_AUTH) < (4, 3):
+        return write_pam_credentials_to_secrets_file(
+            password, overwrite=overwrite, ttl=ttl, _session=ses
+        )
+
+    auth_file = ses.pool.account.derived_auth_file
+    if not auth_file:
+        msg = "Auth file could not be written because no iRODS client environment was found."
+        raise RuntimeError(msg)
+    if ttl:
+        ses.set_auth_option_for_scheme(
+            "pam_password", irods.auth.pam_password.AUTH_TTL_KEY, ttl
+        )
+    ses.set_auth_option_for_scheme(
+        "pam_password", irods.auth.FORCE_PASSWORD_PROMPT, io.StringIO(password)
+    )
+    ses.set_auth_option_for_scheme(
+        "pam_password", irods.auth.STORE_PASSWORD_IN_MEMORY, True
+    )
+    L = []
+    ses.set_auth_option_for_scheme(
+        "pam_password", irods.auth.CLIENT_GET_REQUEST_RESULT, L
+    )
+    with ses.pool.get_connection() as conn:
+        _write_encoded_auth_value(auth_file, L[0], overwrite)
```

Comment:
Is `conn` being used here?

Reply:
Good eye. Yes, we can omit the variable.